aptos-move/move-examples/post_mint_reveal_nft/sources/whitelist.move

```diff
@@ -0,0 +1,145 @@
+        vector::length(&borrow_global<WhitelistMintConfig>(module_address).whitelist_configs)
+    }
+
+    #[view]
+    /// Checks if the given address is whitelisted in the specified whitelist stage.
+    /// Return value u64 is the minting price of the current stage.
+    /// Return value bool is true if the given address is whitelisted.
+    public fun is_user_eligible_for_minting(module_address: address, minter_address: address, stage: u64): (u64, bool) acquires WhitelistMintConfig {
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(module_address);
+        assert!(stage < vector::length(&whitelist_mint_config.whitelist_configs), error::invalid_argument(EINVALID_STAGE));
+        let whitelist_stage = vector::borrow_mut(&mut whitelist_mint_config.whitelist_configs, stage);
+        let user_address_on_whitelist = bucket_table::contains(&whitelist_stage.whitelisted_address, &minter_address);
+        (whitelist_stage.whitelist_mint_price, user_address_on_whitelist)
+    }
+
+    /// Initializes the WhitelistMintConfig resource.
+    public fun init_whitelist_config(admin: &signer) {
+        let config = WhitelistMintConfig {
+            whitelist_configs: vector::empty<WhitelistStage>(),
+        };
+        move_to(admin, config);
+    }
+
+    /// Deducts the number of source certificate that the user wants to mint from the user's minting limit.
+    public(friend) fun deduct_user_minting_amount(module_address: address, minter_address: address, stage: u64, user_minting_amount: u64) acquires WhitelistMintConfig {
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(module_address);
+        assert!(stage < vector::length(&whitelist_mint_config.whitelist_configs), error::invalid_argument(EINVALID_STAGE));
+        let whitelist_stage = vector::borrow_mut(&mut whitelist_mint_config.whitelist_configs, stage);
+        assert!(bucket_table::contains(&whitelist_stage.whitelisted_address, &minter_address), error::permission_denied(EACCOUNT_NOT_WHITELISTED));
+        let remaining_minting_amount = bucket_table::borrow_mut(&mut whitelist_stage.whitelisted_address, minter_address);
+        assert!(*remaining_minting_amount >= user_minting_amount, error::invalid_argument(EEXCEEDS_MINT_LIMIT));
+        *remaining_minting_amount = *remaining_minting_amount - user_minting_amount;
+    }
+
+    /// Adds a new whitelist stage.
+    public fun add_whitelist_stage(admin: &signer, whitelist_start_time: u64, whitelist_end_time: u64, whitelist_price: u64, whitelist_stage: u64) acquires WhitelistMintConfig {
+        assert!(whitelist_start_time < whitelist_end_time, error::invalid_argument(EINVALID_WHITELIST_SETTING));
+        if (!whitelist_config_exists(signer::address_of(admin))) {
+            init_whitelist_config(admin);
+        };
+        let num_stages = get_num_of_stages(signer::address_of(admin));
+        assert!(whitelist_stage <= num_stages, error::invalid_argument(EINVALID_STAGE));
```

Comment:
Why do we need this check? Similar to a previous comment, we can just append the new whitelist config at the end